tests/test_common_view_logic.py

```diff
@@ -1,389 +1,174 @@
-            "articles": [
-                {
-                    "author": "test_author",
-                    "categories": [1, 2],
-                    "display_category": "test_category",
-                    "featured_media": "test",
-                    "group": "test_group",
-                    "image": "test_image",
-                    "tags": ["test"],
-                },
-                {
-                    "author": "test_author",
-                    "categories": [2, 3],
-                    "display_category": "test_category",
-                    "featured_media": "test2",
-                    "group": "test_group",
-                    "image": "test_image",
-                    "tags": ["test2"],
-                },
-            ],
-            "groups": {1: "test_group"},
-            "upcoming": [],
-            "used_categories": {
-                1: "test_category",
-                2: "test_category",
-                3: "test_category",
-            },
-        }
-        self.maxDiff = None
-        self.assertEqual(context, expected_context)
-
-    @patch("canonicalwebteam.blog.wordpress_api.get_group_by_id")
-    @patch("canonicalwebteam.blog.wordpress_api.get_category_by_id")
-    @patch("canonicalwebteam.blog.wordpress_api.get_user")
-    @patch("canonicalwebteam.blog.wordpress_api.get_media")
-    def test_building_index_context_with_featured_articles(
-        self, get_media, get_user, get_category_by_id, get_group_by_id
-    ):
-        get_media.return_value = "test_image"
-        get_user.return_value = "test_author"
-        get_category_by_id.return_value = "test_category"
-        get_group_by_id.return_value = "test_group"
-        articles = [
-            {
-                "featured_media": "test",
-                "author": "test",
-                "categories": [1, 2],
-                "group": [1],
-                "tags": ["test"],
-            },
-            {
-                "featured_media": "test2",
-                "author": "test2",
-                "categories": [2, 3],
-                "group": [1],
-                "tags": ["test2"],
-            },
+    @vcr.use_cassette("fixtures/vcr_cassettes/get_complete_article.yaml")
+    def test_get_complete_article(self):
+        article = api.get_article(
+            "canonical-announces-support-for"
+            "-ubuntu-on-windows-subsystem-for-linux-2"
+        )
+        article["categories"] = [1453]
+
+        complete_article = get_complete_article(article)
+        self.assertEqual(complete_article["author"]["name"], "Canonical")
+        self.assertEqual(
+            complete_article["image"]["id"], article["featured_media"]
+        )
+        self.assertEqual(complete_article["date"], "6 May 2019")
+
+        # The maximum allowed character count for the excerpt is 339
+        # characters. If this limit is reached [...] is added
+        # which is why we test for a max length of 339 + 5 = 344 here
```

Comment:
here we reach the limitations of the library. So this comment makes sense.

Maybe since it's an "edge" case we could unittest it. But not part of this PR anyway